src/memory-fs.js

```diff
@@ -0,0 +1,199 @@
+const path = require('path');
+
+const { createHash } = require(`crypto`);
+const errors = require('errno');
+const util = require('util');
+const { ReadableStream, WritableStream } = require('stream');
+const MemoryFileSystemError = require('memory-fs/lib/MemoryFileSystem');
+
+const debug = util.debuglog('astroturf:memory-fs');
+
+const returnsTrue = () => true;
+const returnsFalse = () => false;
+
+const md5 = input => {
+  const hash = createHash('md5');
+  hash.update(input);
+  return hash.digest('hex');
+};
+
+const read = (file, optsOrEncoding) => {
+  const encoding = optsOrEncoding?.encoding || optsOrEncoding;
+
+  file.atime = new Date();
+
+  return encoding ? file.contents.toString(encoding) : file.contents;
+};
+
+/**
+ * A simple in memorry ponyfill of the node fs that stores objects in memory.
+ * We don't use `memory-fs` because our use-case is much narrower (only need files not dirs)
+ * and it doesn't support stat timestamps
+ */
+class MemoryFs {
+  constructor() {
+    this.paths = new Map();
+
+    ['stat', 'readdir', 'rmdir', 'unlink', 'readFile', 'writeFile'].forEach(
+      fn => {
+        this[fn] = (...args) => {
+          const cb = args.pop();
+          let result;
+          try {
+            result = this[`${fn}Sync`](...args);
+          } catch (err) {
+            setImmediate(() => cb(err));
+            return;
+          }
+          setImmediate(() => cb(null, result));
+        };
+      },
+    );
+  }
+
+  addFile = (p, data, updateMtime = false) => {
+    p = path.normalize(p);
+    const hash = md5(data);
```

Comment:
you sort of don't need a hash... you can just compare the data

Reply:
i know, the only thing is that `data` _may_ be, but problably isn't a buffer vs a string. Theoretically no one should be touching the extracted files, but they can if they need to via `writeFile`. It's hard to know if that ever happens tho. 

maybe it's still cheaper to `toString()` the buffer if it is a buffer to compare?

Reply:
actually I do think it's easier to do create the buffer up front. Otherwise it's messy dealing with the encoding options o the various read method. the original `memory-fs` also does this so it's not likely to be less performant